server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -558,6 +555,56 @@ private IndexMetadata getIndexMetadata(String clusterName, String clusterUUID, U
         }
     }
 
+    /**
+     * Fetch latest metadata from remote cluster state including global metadata and index metadata
+     *
+     * @param clusterUUID uuid of cluster state to refer to in remote
+     * @param clusterName name of the cluster
+     * @return {@link IndexMetadata}
+     */
+    public Metadata getLatestMetadata(String clusterName, String clusterUUID) throws IOException{
+        start();
+        Optional<ClusterMetadataManifest> clusterMetadataManifest = getLatestClusterMetadataManifest(clusterName, clusterUUID);
+        if (!clusterMetadataManifest.isPresent()) {
+            throw new IllegalStateException("Latest index metadata is not present for the provided clusterUUID");
+        }
+        // Fetch Global Metadata
+        Metadata metadataWithGlobalMetadata = getGlobalMetadata(clusterName, clusterUUID, clusterMetadataManifest.get());
+
+        // Fetch Index Metadata
+        Map<String, IndexMetadata> indices = getIndexMetadataMap(clusterName, clusterUUID, clusterMetadataManifest.get());
+
+        return Metadata.builder(metadataWithGlobalMetadata).indices(indices).build();
+    }
+
+    private Metadata getGlobalMetadata(String clusterName, String clusterUUID, ClusterMetadataManifest clusterMetadataManifest) {
+        String globalMetadataFileName = clusterMetadataManifest.getGlobalMetadataFileName();
+        try {
+            // Fetch Global metadata
+            if(globalMetadataFileName != null) {
+                String[] splitPath = globalMetadataFileName.split("/");
+                return BlobStoreRepository.GLOBAL_METADATA_FORMAT.read(
```

Comment:
in https://github.com/opensearch-project/OpenSearch/pull/10404 we have defined a new `GLOBAL_METADATA_FORMAT` . Here we are using `BlobStoreRepository.GLOBAL_METADATA_FORMAT` , both have different naming format.

Reply:
We will use the `GLOBAL_METADATA_FORMAT` defined in upload PR. I will change this as well once that PR is merged.